exercises/practice/sieve/.approaches/nested-loops/content.md

```diff
@@ -0,0 +1,47 @@
+# Nested Loops
+
+```python
+def primes(number):
+    not_prime = []
+    prime = []
+    
+    for item in range(2, number+1):
+        if item not in not_prime:
+            prime.append(item) 
+            for element in range (item*item, number+1, item):
+                not_prime.append(element)
+    
+    return prime
+```
+
+This is the type of code that many people might write as a first attempt.
+
+It is very readable and passes the tests.
+
+The clear disadvantage is that run time is quadratic in the input size: `O(n**2)`, so this approach scales poorly to large input values.
+
+Part of the problem is the line `if item not in not_prime`, where `not-prime` is a list that may be long and unsorted.
+
+This operation requires searching the entire list, so run time is linear in list length: not ideal within a loop repeated many times.
+
+```python
+def primes(number):
+    number += 1
+    l = [True for i in range(number)]
+    for i in range(2, number):
+        if not l[i]:
+            continue
+        for j in range(2 * i, number, i):
+            l[j] = False
+    return [i for i, v in enumerate(l) if i > 1 and v]
+```
```

Comment:
```suggestion
```python
def primes(number):
    number += 1
    prime = [True for item in range(number)]
    for index in range(2, number):
        if not prime[index]:
            continue
        for candidate in range(2 * index, number, index):
            prime[candidate] = False
    return [index for index, value in enumerate(prime) if index > 1 and value]
```
```